src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -49,8 +51,20 @@ class IqPolicyEvaluatorUtil
           new IqClientFactoryConfiguration(credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent,
               log: loggerBridge))
 
-      def verified = iqClient.verifyOrCreateApplication(applicationId)
-      checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
+      try {
+        def verified = iqClient.verifyOrCreateApplication(applicationId)
+        checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
+      }
+      catch (IqClientException e) {
```

Comment:
I would move the error handling to DefaultIqClient.verifyOrCreateApplication. I think it will be simpler there.

Reply:
>I would move the error handling to DefaultIqClient.verifyOrCreateApplication. I think it will be simpler there.

I think that might be a good idea as well.  We have these api calls that basically remove the need for http response handling, but now we have to go into the response and pull out the status code.  It would be nice to push this higher up so that we can make plain `iqClient.verifyOrCreateApplication` calls

Reply:
> I would move the error handling to DefaultIqClient.verifyOrCreateApplication. I think it will be simpler there.

I do believe the implementation for verifyOrCreateApplication is located outside of the jenkins plugin.  It is located in another project.  I think its the java-api project.  Also, all the other calls throw an IqClientException and the plugin looks into the throwables to see what the underlying exception is.  I am not saying that is a good thing.  I just feel changing the way exceptions are handled which are generated from the java-api project is outside of the scope of this PR.

Reply:
> I am not saying that is a good thing. I just feel changing the way exceptions are handled which are generated from the java-api project is outside of the scope of this PR.

Personally, I'll take that as a valid answer. If we have to modify that much to change this message I don't think that's something that was anticipated when we sized this, and even if it's a trivial change, it does add a lot more time in terms of completing the story.

Reply:
> I was thinking that we could move the exception handling to DefaultIqClient.verifyOrCreateApplication, which will throw now an IqClientException with the new message. 

That sounds good to me.  I misunderstood your suggestion before.  But now I understand what you are saying and I like it.  I would add this message to the DefaultIqclient and I wouldn't need to change anything in the Jenkins project besides updating the POM.  Jenkins already has the ability to show the exception.